src/map/npc.c

```diff
@@ -2826,73 +2825,65 @@ static int npc_selllist(struct map_session_data *sd, struct itemlist *item_list)
 	if (sd->status.zeny >= MAX_ZENY && nd->master_nd == NULL)
 		return 1;
 
-	// verify the sell list
+	// Verify the sell list
 	for (i = 0; i < VECTOR_LENGTH(*item_list); i++) {
 		struct itemlist_entry *entry = &VECTOR_INDEX(*item_list, i);
 		int nameid, value, idx = entry->id;
 
-		if (idx >= sd->status.inventorySize || idx < 0 || entry->amount < 0) {
+		if (idx >= sd->status.inventorySize || idx < 0 || entry->amount < 0)
 			return 1;
-		}
 
-		if (duplicates[idx]) {
-			// Sanity check. The client sends each inventory index at most once [Haru]
+		if (duplicates[idx]) // Sanity check. The client sends each inventory index at most once [Haru]
 			return 1;
-		}
+
 		duplicates[idx] = true;
 
 		nameid = sd->status.inventory[idx].nameid;
 
-		if (!nameid || !sd->inventory_data[idx] || sd->status.inventory[idx].amount < entry->amount) {
+		if (nameid == 0 || sd->inventory_data[idx] == 0 || sd->status.inventory[idx].amount < entry->amount)
```

Comment:
Nitpick: The check for `sd->inventory_data[idx]` should be against NULL instead of 0 (since it's a pointer)

Reply:
Eagle eye...
*fixed*

Reply:
Do you mean by skipping the verification entirely for script controlled shops? Verification of the item list to make sure that it isn't malformed/forged is still necessary for those shops, isn't it?